### PR TITLE
Pullrequest code cleanup weaken types

### PR DESCRIPTION
**Pull request by BenSartor**

### CODE REVIEW

File: app/src/main/java/org/simlar/widgets/ContactsAdapter.java [hunk 1]
```diff
@@ -133,7 +134,7 @@ public ContactsAdapter(final Context context)
 		return rowView;
 	}
 
-	public void addAllContacts(final Set<ContactDataComplete> contacts)
+	public void addAllContacts(final Collection<ContactDataComplete> contacts)
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/widgets/ContactsAdapter.java]: Semantic change! We do not want duplicates here.

File: app/src/main/java/org/simlar/widgets/RingingActivity.java [hunk 2]
```diff
@@ -39,11 +39,12 @@
 import org.simlar.utils.Util;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 
 public final class RingingActivity extends AppCompatActivity
 {
-	private final List<View> mCircles = new ArrayList<>();
+	private final Collection<View> mCircles = new ArrayList<>();
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/widgets/RingingActivity.java]: We really do want the circles in their correct order here.

File: app/src/main/java/org/simlar/helper/SimlarNumber.java [hunk 3]
```diff
@@ -157,7 +158,7 @@ public static String createSimlarId(final String telephoneNumber)
 		return new SimlarNumber(telephoneNumber).getSimlarId();
 	}
 
-	public static Set<Integer> getSupportedCountryCodes()
+	public static Collection<Integer> getSupportedCountryCodes()
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/SimlarNumber.java]: Semantic change! We do not want duplicates here.

File: app/src/main/java/org/simlar/helper/SimlarNumber.java [hunk 4]
```diff
@@ -160,7 +160,7 @@ public static String createSimlarId(final String telephoneNumber)
 
 	public static Collection<Integer> getSupportedCountryCodes()
 	{
-		final Set<Integer> supportedCountryCodes = new HashSet<>();
+		final Collection<Integer> supportedCountryCodes = new HashSet<>();
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/SimlarNumber.java]: We do not want duplicates here.

File: app/src/main/java/org/simlar/helper/PermissionsHelper.java [hunk 5]
```diff
@@ -130,7 +131,7 @@ public static void requestMajorPermissions(final Activity activity, final boolea
 	private static boolean checkAndRequestPermissions(final Activity activity, final Set<Type> types)
 	{
 		final Set<Type> requestTypes = EnumSet.noneOf(Type.class);
-		final Set<String> rationalMessages = new HashSet<>();
+		final Collection<String> rationalMessages = new HashSet<>();
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/PermissionsHelper.java]: We do not want duplicates here.

File: app/src/main/java/org/simlar/contactsprovider/ContactsProvider.java [hunk 6]
```diff
@@ -88,7 +89,7 @@ private ContactsProvider()
 		private Map<String, ContactData> mContacts = new HashMap<>();
 		private State mState = State.UNINITIALIZED;
 		boolean mFakeData = false;
-		private final Set<FullContactsListener> mFullContactsListeners = new HashSet<>();
+		private final Collection<FullContactsListener> mFullContactsListeners = new HashSet<>();
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/contactsprovider/ContactsProvider.java]: We do not want duplicates here.

File: app/src/main/java/org/simlar/helper/CallConnectionDetails.java [hunk 7]
```diff
@@ -129,12 +129,12 @@ public int getQualityDescription()
 		return mQuality.getDescription();
 	}
 
-	public String getCodec()
+	public CharSequence getCodec()
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/CallConnectionDetails.java]: CharSequence "does not refine the general contracts of the equals and hashCode methods. The result of comparing two objects that implement CharSequence is therefore, in general, undefined." We want comparable strings, in general, even if we do not actually compare them, yet.

**Comment by BenSartor** [comment]
As we dropped most of the weaken type commit, I created a new pullrequest #292.

**Pull request closed without merging**